### ISSUE TITLE
DualStack Openshift causes the memcached listen IPs to be wrongly formatted

### ISSUE THREAD

**Issue by blackphantom39** [openstack-k8s-operators/infra-operator#233]
Hello,

I'm currently testing the 18.0-Beta of the RHOSO deployment and ran into the following issue creating the control plane and therefore the memcached pods.

Given the PodIPs look like this:

```yaml
  podIPs:
    - ip: 10.130.1.23
    - ip: 'fd74:ae5d:e44d:1::1826'
```

the resulting environment variable for POD_IPS will look like: `POD_IPS=10.130.0.198,fd74:ae5d:e44d:1::19b5` this will not work correctly when inserted in the [template](https://github.com/openstack-k8s-operators/infra-operator/blob/main/templates/memcached/config/memcached) for the LISTEN_IPS. The resulting LISTEN variable will have a malformed IPv6 causing the pod to crash.

Resulting LISTEN:

```
echo $LISTEN
127.0.0.1 notls:127.0.0.1:11211 [::1] notls:[::1]:11211 [10.130.0.198,fd74:ae5d:e44d:1::19b5] notls:[10.130.0.198,fd74:ae5d:e44d:1::19b5]:11211
                                                        └─────── not correctly split ───────┘
```

**Comment by blackphantom39** [comment]
It should be sufficient to just replace the reference to $POD_IPS by either of the following:

`$(printf %s $POD_IPS | tr ',' ' ')` resulting in:
```
LISTEN=$(echo 127.0.0.1 ::1 $(printf %s $POD_IPS | tr ',' ' ') | tr ' ' '\n' | sed 's/\(.*\):\(.*\)/[\1:\2]/' {{ .memcachedTLSListen }})
```

or  `$(echo $POD_IPS | tr ',' ' ')` resulting in:
```
LISTEN=$(echo 127.0.0.1 ::1 $(echo $POD_IPS | tr ',' ' ') | tr ' ' '\n' | sed 's/\(.*\):\(.*\)/[\1:\2]/' {{ .memcachedTLSListen }})
```

**Comment by stuggi** [comment]
@blackphantom39 thanks for reporting this bug. you want to submit a PR to fix this?

**Comment by stuggi** [comment]
just fyi, there will probably be follow up issues in other components as for GA dual stack was not targeted as a supported feature.

**Comment by blackphantom39** [comment]
Closed as of PR #236
